components/postcode-lookup/index.js

```diff
@@ -0,0 +1,339 @@
+        postcodeError = conditionalTranslate(key, req.translate) ||
+          defaults.POSTCODE_ERROR[postcodeApiMessageKey];
+      }
+
+      let addressCount = '';
+      if (req.sessionModel.get('addressCount') === 1) {
+        addressCount = '1 address for ';
+      } else {
+        addressCount = req.sessionModel.get('addressCount') + ' addresses for ';
+      }
+
+      return _.merge({}, locals, {
+        enterManually: conditionalTranslate('pages.address.lookup.enter-manually', req.translate) ||
+          defaults.ENTER_MANUALLY,
+        postcodeHeading: conditionalTranslate('pages.postcode.postcode-heading', req.translate) ||
+          defaults.POSTCODE_HEADING,
+        addressLookupHeading: conditionalTranslate('pages.address-lookup.address-lookup-heading', req.translate) ||
+          defaults.ADDRESS_LOOKUP_HEADING,
+        manualAddressHeading: conditionalTranslate('pages.address-lookup.manual-address-heading', req.translate) ||
+          defaults.MANUAL_ADDRESS_HEADING,
+        manualAddressParagraph: conditionalTranslate('pages.address-lookup.manual-address-paragraph', req.translate) ||
+          defaults.MANUAL_ADDRESS_PARAGRAPH,
+        searchErrorHeading: conditionalTranslate('pages.postcode-search-problem.searchErrorHeading', req.translate) ||
+          defaults.SEARCH_ERROR_HEADING,
+        postcodeEntered: sessionPostcode,
+        addressesExist: (req.sessionModel.get('addresses') !== undefined &&
+        req.sessionModel.get('addresses').length > 0) ? true : false,
+        noAddressHeading: conditionalTranslate('pages.postcode-search-problem.noAddressHeading', req.translate) ||
+          defaults.NO_ADDRESS_HEADING,
+        addressCount: addressCount,
+        postcodeSearchTitle: conditionalTranslate('pages.postcode-search.title', req.translate) ||
+          defaults.ADDRESS_LOOKUP_SEARCH_TITLE,
+        addressLookupTitle: conditionalTranslate('pages.address-lookup.title', req.translate) ||
+          defaults.ADDRESS_LOOKUP_TITLE,
+        addressDetailsTitle: conditionalTranslate('pages.address-details.title', req.translate) ||
+          defaults.ADDRESS_DETAILS_TITLE,
+        lookupProblemTitle: conditionalTranslate('pages.address-lookup-problem.title', req.translate) ||
+          defaults.ADDRESS_LOOKUP_PROBLEM_TITLE,
+        route: this.options.route,
+        editLink,
+        cantFind,
+        searchByPostcodeLink,
+        postcodeError,
+        section
+      });
+    }
+
+    postcode(req, res, callback) {
+      // Clear the value stored in the addresses radio button group
+      req.sessionModel.set(`${addressKey}-select`, '');
+      // Call OS Places API to return list of addresses by postcode
+      const enteredPostcode = req.form.values[`${addressKey}-postcode`];
+
+      axios.get(apiURL + '?postcode=' +  enteredPostcode + '&key=' + apiKey)
+        .then(function (response) {
+          const addressList = [];
+          _.forOwn(response.data.results, function (value) {
```

Comment:
do some checks on the response, make sure it is of the expected structure & type

Reply:
Added a check for the 'data' property in the response, but didn't check for any sub-properties of that because the component needs to still go to the address lookup page when there are no results returned for a given postcode.